src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -0,0 +1,139 @@
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_PHONE.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_PHONE)));
+        } else if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_EMAIL.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_EMAIL)));
+        } else if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_ADDRESS.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_ADDRESS)));
+        } else {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_TAG.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_TAG)));
+        }
+    }
+
+    private Pair<List<String>, List<List<String>>> extractPrefixesAndKeywords(ArgumentMultimap argMultimap)
+            throws ParseException {
+        List<String> prefixes = new ArrayList<>();
+        List<List<String>> keywords = new ArrayList<>();
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            prefixes.add(PREFIX_NAME.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_NAME));
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            prefixes.add(PREFIX_PHONE.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_PHONE));
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            prefixes.add(PREFIX_EMAIL.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_EMAIL));
+        }
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            prefixes.add(PREFIX_ADDRESS.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_ADDRESS));
+        }
+        if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
+            prefixes.add(PREFIX_TAG.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_TAG));
+        }
+        if (prefixes.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+        }
+        return new Pair<>(prefixes, keywords);
+    }
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>} if {@code tags} is non-empty.
+     * If {@code tags} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Tag>} containing zero tags.
+     */
+    private Optional<Set<Tag>> parseTagsForSearch(Collection<String> tags) throws ParseException {
+        assert tags != null;
+
+        if (tags.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> tagSet = tags.size() == 1 && tags.contains("") ? Collections.emptySet() : tags;
+        return Optional.of(ParserUtil.parseTags(tagSet));
+    }
+
+}
```

Comment:
Yes it is just a placeholder method, not yet finished haha. At the beginning I tried to copy and modify the `parseTagsForEdit` method but then realize that it is better to leave implementation for search tags first until the tags feature is finalized. So I just left it there in case I really turn out to be needing that in the future 😂. Thanks for pointing that out!